### PR TITLE
[DOC] Use simple table ReST syntax for tables in API docs

### PR DESCRIPTION
**Pull request by rahulporuri**
Similar to #5564, this PR updates the (mostly broken) ReST table syntax in API documentation. In addition, this PR also fixes ReST issues of the sort "Title underline too short", seen in the logs when building docs.

#### Reference Issues/PRs
N/A

#### What does this implement/fix? Explain your changes.

Update docstrings to use the [ReST simple tables syntax](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#simple-tables)

#### Does your contribution introduce a new dependency? If yes, which one?

No

#### What should a reviewer concentrate their feedback on?

Confirm that the API docs look correct from the docs built on this branch

#### Did you add any tests for the change?

N/A

#### Any other comments?

See the fixed API docs at

- [`RockerRegressor`](https://sktime--5595.org.readthedocs.build/en/5595/api_reference/auto_generated/sktime.regression.kernel_based.RocketRegressor.html#rocketregressor)
- [`TSFreshFeatureExtractor.fit_transform`](https://sktime--5595.org.readthedocs.build/en/5595/api_reference/auto_generated/sktime.transformations.panel.tsfresh.TSFreshFeatureExtractor.html#sktime.transformations.panel.tsfresh.TSFreshFeatureExtractor.fit_transform)

Note that the `BaseTransformer` doesn't seem to be included in the API docs and the `SupervisedTransformer` link in the [Estimator overview table](https://sktime--5595.org.readthedocs.build/en/5595/estimator_overview.html) is broken.

##### For all contributions
- [ ] ~I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)~
- [ ] ~Optionally, I've added myself and possibly others to the [CODEOWNERS](https://github.com/sktime/sktime/blob/main/CODEOWNERS) file - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).~
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [ ] ~I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.~
- [ ] ~I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.~
- [ ] ~If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).~

### CODE REVIEW

**Comment by fkiraly** [comment]
this is how it looks like - does not seem quite right:

image

Should it not be a table?

I thought this was fine:

```

        .. list-table::
            :widths: 35 35 40
            :header-rows: 2
            * -
              - `transform`
              -
            * - `X`
              - `-output`
              - type of return
            * - `Series`
              - `Primitives`
              - `pd.DataFrame` (1-row)
            * - `Panel`
              - `Primitives`
              - `pd.DataFrame`
            * - `Series`
              - `Series`
              - `Series`
            * - `Panel`
              - `Series`
              - `Panel`
            * - `Series`
              - `Panel`
              - `Panel`
```

(it results in an rst table)

**Comment by rahulporuri** [comment]
> (it results in an rst table)

@fkiraly it does result in a ReST table but that ReST table is only easily readable in the built docs. The code itself isn't as readable, in my humble opinion.

But if this isn't the look you're all going for, i'm okay with using the `list-table` style ReST table.

File: docs/source/estimator_overview_table.md
Comment (yarnabrina) [inline docs/source/estimator_overview_table.md:1]: Can you please explain addition of this new file? What are you trying to add?

Also, how is this going to interact with lines 373-374 in `conf.py` file?

https://github.com/sktime/sktime/blob/df21a0c0275ebf28deb30efac5d469c9f0d178e3/docs/source/conf.py#L373-L374

**Review by fkiraly — CHANGES_REQUESTED** [review]
>@fkiraly it does result in a ReST table but that ReST table is only easily readable in the built docs. The code itself isn't as readable, in my humble opinion.

I know, but the rst docs also do not look that great.

How about this compromise:

* in public methods, we use rest syntax
* in private methods (which is what someone looking at estimator code sees), we use sth that looks like a table?

Or, we repeat the table as a comment?

**Pull request closed without merging**